### PR TITLE
kinder: enable the v1beta4 API for >= 1.31

### PR DESCRIPTION
**Pull request by neolit123**
requires https://github.com/kubernetes/kubernetes/pull/125029

xref https://github.com/kubernetes/kubeadm/pull/3067

### CODE REVIEW

**Comment by pacoxu** [comment]
```
W0617 10:43:17.066823     200 initconfiguration.go:332] error unmarshaling configuration schema.GroupVersionKind{Group:"kubeadm.k8s.io", Version:"v1beta4", Kind:"InitConfiguration"}: json: cannot unmarshal object into Go struct field NodeRegistrationOptions.nodeRegistration.kubeletExtraArgs of type []v1beta4.Arg
json: cannot unmarshal object into Go struct field NodeRegistrationOptions.nodeRegistration.kubeletExtraArgs of type []v1beta4.Arg
Error: failed to exec action kubeadm-init: exit status 1
 exit status 1
```
The CI failure is valid.

File: kinder/pkg/kubeadm/config.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ nodeRegistration:
     node-ip: "{{ .NodeAddress }}"
 ---
 # no-op entry that exists solely so it can be patched
-apiVersion: kubeadm.k8s.io/v1beta2
+apiVersion: kubeadm.k8s.io/v1beta4
```
Comment (pacoxu) [inline kinder/pkg/kubeadm/config.go:173]: With https://github.com/kubernetes/kubernetes/pull/119156, 
```
  kubeletExtraArgs:
    node-ip: "{{ .NodeAddress }}"
```

should be 

```
  kubeletExtraArgs:
  - name: node-ip
     value: "{{ .NodeAddress }}"
```

Comment (neolit123) [reply kinder/pkg/kubeadm/config.go:177]: fixed

**Pull request merged**